server.py

```diff
@@ -381,6 +382,112 @@ def tilejson(fmt, tile_pixel_size=None):
     return resp
 
 
+def t2_meta_and_offset(requested_tile, materialized_zooms, metatile_size):
+    # Find the materialized zoom that holds this tile
+    mz = next(x for x in sorted(materialized_zooms, reverse=True) if x <= requested_tile.z)
+
+    # Find the tile at the materialized zoom that holds the requested tile
+    dz = requested_tile.z - mz
+    mx = requested_tile.x >> dz
+    my = requested_tile.y >> dz
+    mx -= mx % metatile_size
+    my -= my % metatile_size
+
+    meta = TileRequest(mz, mx, my, 1, 'zip')
+
+    # Build the key for the tile inside the archive
+    offset = requested_tile
+
+    return meta, offset
+
+
+def t2_extract_tile(metatile_bytes, offset):
+    data = BytesIO(metatile_bytes)
+    z = zipfile.ZipFile(data, 'r')
+
+    offset_key = '{zoom}/{x}/{y}{scale}.{format}'.format(
+        zoom=offset.z,
+        x=offset.x,
+        y=offset.y,
+        scale='' if offset.scale < 2 else '@%dx' % offset.scale,
+        format=offset.format,
+    )
+
+    try:
+        return z.read(offset_key)
+    except KeyError as e:
+        raise TileNotFoundInMetatile("Couldn't find tile %s in metatile" % offset_key)
+
+
+def t2_retrieve_tile(meta, offset, cache_info):
+    metatile_data = metatile_fetch(meta, cache_info)
+    tile_data = t2_extract_tile(metatile_data.data, offset)
+
+    return StorageResponse(
+        data=tile_data,
+        cache_info=CacheInfo(
+            last_modified=metatile_data.cache_info.last_modified,
+            etag=metatile_data.cache_info.etag,
+        )
+    )
+
+
+@tile_bp.route('/tilezen/landcover/v1/<int:tile_pixel_size>/all/<int:z>/<int:x>/<int:y>.<fmt>')
+@tile_bp.route('/tilezen/landcover/v1/all/<int:z>/<int:x>/<int:y>.<fmt>')
+def handle_landcover_tile(z, x, y, fmt, tile_pixel_size=None):
+    if not is_valid_tile_request(z, x, y, max_zoom=current_app.config.get('LANDCOVER_MAX_ZOOM')):
+        return abort(400, "Requested tile out of range.")
+
+    tile_pixel_size = tile_pixel_size or 256
+    tile_size = tile_pixel_size / 256
+    if tile_size != int(tile_size):
+        return abort(400, "Invalid tile size. %s is not a multiple of 256." % tile_pixel_size)
+    if tile_size != 2:
+        return abort(400, "Landcover only supports 512 tile size.")
+    tile_size = int(tile_size)
+
+    requested_tile = TileRequest(z, x, y, tile_size, fmt)
+
+    t2_materialized_zooms = [0, 7]
+    t2_metatile_size = 1
```

Comment:
Are `t2_materialized_zooms` and `t2_metatile_size` still used?

Reply:
Nope, deleted.